util/fhir/resourceCreation.ts

```diff
@@ -0,0 +1,152 @@
+      }
+    ],
+    gender,
+    birthDate
+  };
+
+  return JSON.stringify(pt, null, 2);
+}
+
+/**
+ * Creates copies of all passed in resources (without references maintained) and gives them
+ * new resource ids. Replaces all patient references to the patient oldId with newId
+ * @param copyResources {fhir4.FhirResource[]} array of fhir resources to be copied
+ * @param oldId {String} a patient id that the copyResources may reference
+ * @param newId {String} a patient id that should replace oldId in references
+ * @returns {fhir4.FhirResource[]} array of new resource copies
+ */
+export function createCopiedResources(
+  copyResources: fhir4.FhirResource[],
+  oldId: string,
+  newId: string
+): fhir4.FhirResource[] {
+  const resources: fhir4.FhirResource[] = copyResources.map(cr => {
+    let resourceString = JSON.stringify(cr);
+    const idRegexp = new RegExp(`Patient/${oldId}`, 'g');
+    resourceString = resourceString.replace(idRegexp, `Patient/${newId}`);
+    const resource: fhir4.FhirResource = JSON.parse(resourceString);
+    resource.id = uuidv4();
+    // Note: this does not update potential cross-resource references, which we may want to support in the future
+    return resource;
+  });
+  return resources;
+}
+
+/**
+ * Creates a copy of the passed in patient object (without references maintained) and updates the
+ * id and identifier as well as creating a new name to differentiate the new patient copy
+ * @param copyPatient {fhir4.Patient} a fhir Patient object to copy
+ * @returns {fhir4.Patient} the new fhir patient copy
+ */
+export function createCopiedPatientResource(copyPatient: fhir4.Patient): fhir4.Patient {
+  const patient: fhir4.Patient = _.cloneDeep(copyPatient);
+  const identifier = patient.identifier?.find(id => id.system === 'http://example.com/test-id');
+  patient.id = uuidv4();
+  if (identifier) {
+    identifier.value = `test-patient-${patient.id}`;
+  } else {
+    const newIdentifier: fhir4.Identifier = {
+      use: 'usual',
+      system: 'http://example.com/test-id',
+      value: `test-patient-${patient.id}`
+    };
+    if (patient.identifier) {
+      patient.identifier.push(newIdentifier);
+    } else {
+      patient.identifier = [newIdentifier];
+    }
+  }
+  if (patient.name && patient.name.length > 0) {
+    patient.name[0] = {
+      family: getRandomLastName(),
+      given: [getRandomFirstName(patient.gender === 'male' ? 'male' : 'female')] // future should handle non-binary
+    };
+  }
+  return patient;
+}
+
+/**
+ * Creates a string representing a patient bundle resource. Creates using a patient resource and
+ * an array of the patient's associated resources
+ * @param {Object} patient FHIR Patient object
+ * @param {Array} resources array of FHIR resources associated with the patient
+ * @returns {String} representation of a FHIR patient bundle resource
+ */
+export function createPatientBundle(patient: fhir4.Patient, resources: fhir4.FhirResource[]): fhir4.Bundle {
```

Comment:
Kind of similar comment as above, would it make more sense for this to be in `patient.ts`? I could be very wrong :)